loopd/swapclient_server.go

```diff
@@ -210,6 +210,38 @@ func (s *swapClientServer) LoopOut(ctx context.Context,
 		PaymentTimeout:          paymentTimeout,
 	}
 
+	// If the asset id is set, we need to set the asset amount and asset id
+	// in the request.
+	if in.AssetInfo != nil {
+		if in.AssetInfo.AssetId == nil &&
```

Comment:
I propose to check if it is set with `len`:
```go
if len(in.AssetInfo.AssetId) != 0 &&
```

Since an empty slice can be not nil.

And in 2 cases below also.